src/throttler.decorator.ts

```diff
@@ -1,11 +1,26 @@
-import { Inject } from '@nestjs/common';
-import { THROTTLER_LIMIT, THROTTLER_SKIP, THROTTLER_TTL } from './throttler.constants';
-import { getOptionsToken, getStorageToken } from './throttler.providers';
+import { ExecutionContext, Inject } from '@nestjs/common';
 import { Resolvable } from './throttler-module-options.interface';
+import {
+  THROTTLER_KEY_GENERATOR,
+  THROTTLER_LIMIT,
+  THROTTLER_SKIP,
+  THROTTLER_TRACKER,
+  THROTTLER_TTL,
+} from './throttler.constants';
+import { getOptionsToken, getStorageToken } from './throttler.providers';
+
+export type ThrottlerGetTrackerFunction = (req: Record<string, any>) => Promise<string> | string;
+export type ThrottlerGenerateKeyFunction = (
+  context: ExecutionContext,
+  trackerString: string,
+  throttlerName: string,
+) => string;
```

Comment:
Could we move these to the `throttler-module-options.interface` file? That way we can keep the interfaces together.

Reply:
I moved the definitions to the mentioned file. Do they also need such `@publicApi` comments like the other interfaces or are those not relevant for these definitions?

Reply:
Yes, that allows the api doc generator to know to add them to the docs